databricks/sdk/core.py

```diff
@@ -873,7 +873,12 @@ def _known_file_config_loader(self):
             return
         config_file = self.config_file
         if not config_file:
-            config_file = "~/.databrickscfg"
+            if os.environ.get('RUNNING_IN_TEST_MODE') and "testdata" not in os.environ.get('HOME'):
```

Comment:
Make it look for a different path, when this is in test mode and `HOME` is not overwritten.

Not sure if this is the best approach. @mgyucht Let me know if you have better ideas